src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/Metadata/ValidationErrorMessages.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Metadata
+{
+    public abstract class ValidationErrorMessages
+    {
+        /// <summary>
+        /// Error message the model binding system adds when a property with an associated
+        /// <see cref="BindRequiredAttribute"/> is not bound.
+        /// </summary>
+        /// <value>Default <see cref="string"/> is "A value for the '{0}' property was not provided.".</value>
+        public abstract Func<string> MissingBindRequiredValueResource { get; }
+
+        /// <summary>
+        /// Error message the model binding system adds when either the key or the value of a
+        /// <see cref="System.Collections.Generic.KeyValuePair{TKey, TValue}"/> is bound but not both.
+        /// </summary>
+        /// <value>Default <see cref="string"/> is "A value is required.".</value>
```

Comment:
I plan to change this resource and `ValueInvalid_MustNotBeNullResource` to include the key. As-is these are not very useful in a validation summary.